src/app/extensions/punchout/services/punchout/punchout.service.ts

```diff
@@ -299,6 +302,62 @@ export class PunchoutService {
     return cXmlForm;
   }
 
+  /**
+   * Gets the cXML configuration for a given user.
+   *
+   * @param user    The punchout user.
+   * @returns       An array of punchout cXML configurations.
+   */
+  getCxmlConfiguration(user: string): Observable<CxmlConfiguration[]> {
+    return this.currentCustomer$.pipe(
+      switchMap(customer =>
+        this.apiService
+          .get<CxmlConfigurationData>(
+            `customers/${encodeResourceID(customer.customerNo)}/punchouts/${encodeResourceID(
+              this.getResourceType('cxml')
+            )}/users/${user}/configurations`,
+            {
+              headers: new HttpHeaders({
+                Accept: 'application/vnd.intershop.punchout.cxml.v3+json',
+              }),
+            }
+          )
+          .pipe(map(CxmlConfigurationMapper.fromData))
+      )
+    );
+  }
+
+  /**
+   *  Updates a punchout cxml configuration.
+   *
+   * @param cxmlConfiguration  An array of cxml configuration items to update.
+   * @param user               The selected punchout user.
+   * @returns                  The updated cxml configuration.
+   */
+  updateCxmlConfiguration(cxmlConfiguration: CxmlConfiguration[], user: string): Observable<CxmlConfiguration[]> {
+    if (!cxmlConfiguration) {
+      return throwError(() => new Error('updateCxmlConfiguration() called without required CxmlConfiguration'));
+    }
+
+    return this.currentCustomer$.pipe(
+      switchMap(customer =>
+        this.apiService
+          .patch<CxmlConfigurationData>(
+            `customers/${encodeResourceID(customer.customerNo)}/punchouts/${encodeResourceID(
```

Comment:
the encoding of the resourceType is not necessary even this function is used elsewhere